csharp/generate/templates/ApiClient.mustache

```diff
@@ -409,6 +405,8 @@ namespace {{packageName}}.Client
             var transformed = new ApiResponse<T>(response.StatusCode, new Multimap<string, string>({{#caseInsensitiveResponseHeaders}}StringComparer.OrdinalIgnoreCase{{/caseInsensitiveResponseHeaders}}), result, rawContent)
             {
                 ErrorText = response.ErrorMessage,
+                ResponseStatus = response.ResponseStatus,
```

Comment:
What is `reponseStatus`?

Reply:
This is what is assigned to the response object whenever for example a response is a 200, but the deserialization fails because of body and returned API string mismatch. There is no handling for this in its current state, that's why we need to check for ResponseStatus = ResponseStatus.Error in the custom exception factory

Reply:
But if you can think of a better way to handle that, then I'd be interested to hear